packages/core/src/auth/auth.ts

```diff
@@ -170,7 +171,13 @@ export class Auth implements AuthService, ConnectionManager {
         const profile = this.store.getProfileOrThrow(id)
         if (profile.type === 'sso') {
             const provider = this.getSsoTokenProvider(id, profile)
-            await this.authenticate(id, () => provider.createToken())
+
+            // We know createToken() will eventually emit for aws_loginWithBrowser, so we pre-emptively set isReAuth
+            // using the following:
+            await telemetry.aws_loginWithBrowser.run(async span => {
+                span.record({ isReAuth: true })
+                await this.authenticate(id, () => provider.createToken())
+            })
```

Comment:
I think this means the `aws_loginWithBrowser.run()` in `packages/core/src/auth/sso/ssoAccessTokenProvider.ts` will be redundant, or "nested". Not a big deal, but ideally in `ssoAccessTokenProvider.ts` we would skip the redundant `aws_loginWithBrowser.run()` if the current scope/execution already is tagged as `aws_loginWithBrowser`.

Reply:
Also because you are doing `span.record()` instead of `telemetry.record()`, I think the nested/redundant `aws_loginWithBrowser` won't have `isReAuth`.

It's easy to confirm/disconfirm this using the logs (e.g. `AWS (Developer): Watch Logs`).

Reply:
Good catch. When I was testing before I missed the nested metric was also being emitted which is redundant.

I'm going with the `telemetry.activeSpan` check in SsoAccessTokenProvider to check if doing a `run()` is required